synapse/storage/data_stores/main/events_worker.py

```diff
@@ -74,6 +76,26 @@ class EventsWorkerStore(SQLBaseStore):
     def __init__(self, database: Database, db_conn, hs):
         super(EventsWorkerStore, self).__init__(database, db_conn, hs)
 
+        if hs.config.worker_app is None:
```

Comment:
I kinda hate this pattern; it's everywhere and it feels magical. It feels like there should be an `hs.am_I_the_source_for_the_events_stream()` method or something. maybe that's one to punt to a different PR where we can try to kill off other instances of the same thing though.

Reply:
(or maybe you have a better plan for this anyway)

Reply:
This is how we currently do it, but yes configuration for this sort of thing is incoming :)